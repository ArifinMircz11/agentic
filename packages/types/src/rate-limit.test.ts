import { expect, expectTypeOf, test } from 'vitest'

import {
  type RateLimit,
  type RateLimitInput,
  rateLimitSchema
} from './rate-limit'

test('rateLimitSchema valid', () => {
  expect(
    rateLimitSchema.parse({
      interval: 10,
      maxPerInterval: 100
    })
  ).toMatchSnapshot()

  expect(
    rateLimitSchema.parse({
      interval: '10s',
      maxPerInterval: 100
    })
  ).toMatchSnapshot()

  expect(
    rateLimitSchema.parse({
      interval: '1 day',
      maxPerInterval: 1000
    })
  ).toMatchSnapshot()
})

test('rateLimitSchema invalid', () => {
  expect(() =>
    rateLimitSchema.parse({
      interval: '',
      maxPerInterval: 5
    })
  ).toThrowErrorMatchingSnapshot()

  expect(() =>
    rateLimitSchema.parse({
      interval: 0,
      maxPerInterval: 5
    })
  ).toThrowErrorMatchingSnapshot()

  expect(() =>
    rateLimitSchema.parse({
      interval: '--',
      maxPerInterval: 10
    })
  ).toThrowErrorMatchingSnapshot()

  expect(() =>
    rateLimitSchema.parse({
      interval: '1 day',
      maxPerInterval: -1000
    })
  ).toThrowErrorMatchingSnapshot()
})

test('RateLimit types', () => {
  expectTypeOf({
    interval: 10,
    maxPerInterval: 100
  } as const).toExtend<RateLimit>()

  expectTypeOf<{
    interval: 10
    maxPerInterval: 100
  }>().toExtend<RateLimit>()

  expectTypeOf({
    interval: '10s',
    maxPerInterval: 100
  } as const).not.toExtend<RateLimit>()

  expectTypeOf<{
    interval: '10s'
    maxPerInterval: 100
  }>().not.toExtend<RateLimit>()
})

test('RateLimitInput types', () => {
  expectTypeOf({
    interval: 10,
    maxPerInterval: 100
  } as const).toExtend<RateLimitInput>()

  expectTypeOf<{
    interval: 10
    maxPerInterval: 100
  }>().toExtend<RateLimitInput>()

  expectTypeOf({
    interval: '3h',
    maxPerInterval: 100
  } as const).toExtend<RateLimitInput>()

  expectTypeOf<{
    interval: '3h'
    maxPerInterval: 100
  }>().toExtend<RateLimitInput>()
})
