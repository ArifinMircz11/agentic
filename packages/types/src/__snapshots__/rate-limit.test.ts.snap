// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rateLimitSchema invalid 1`] = `
[ZodError: [
  {
    "code": "invalid_union",
    "unionErrors": [
      {
        "issues": [
          {
            "code": "invalid_type",
            "expected": "number",
            "received": "string",
            "path": [
              "interval"
            ],
            "message": "Expected number, received string"
          }
        ],
        "name": "ZodError"
      },
      {
        "issues": [
          {
            "code": "too_small",
            "minimum": 1,
            "type": "string",
            "inclusive": true,
            "exact": false,
            "message": "String must contain at least 1 character(s)",
            "path": [
              "interval"
            ]
          }
        ],
        "name": "ZodError"
      }
    ],
    "path": [
      "interval"
    ],
    "message": "Invalid input"
  }
]]
`;

exports[`rateLimitSchema invalid 2`] = `
[ZodError: [
  {
    "code": "too_small",
    "minimum": 0,
    "type": "number",
    "inclusive": false,
    "exact": false,
    "message": "Number must be greater than 0",
    "path": [
      "interval"
    ]
  }
]]
`;

exports[`rateLimitSchema invalid 3`] = `
[ZodError: [
  {
    "code": "custom",
    "message": "Invalid interval \\"--\\"",
    "path": [
      "interval",
      "interval"
    ]
  }
]]
`;

exports[`rateLimitSchema invalid 4`] = `
[ZodError: [
  {
    "code": "too_small",
    "minimum": 0,
    "type": "number",
    "inclusive": true,
    "exact": false,
    "message": "Number must be greater than or equal to 0",
    "path": [
      "maxPerInterval"
    ]
  }
]]
`;

exports[`rateLimitSchema valid 1`] = `
{
  "interval": 10,
  "maxPerInterval": 100,
}
`;

exports[`rateLimitSchema valid 2`] = `
{
  "interval": 10,
  "maxPerInterval": 100,
}
`;

exports[`rateLimitSchema valid 3`] = `
{
  "interval": 86400,
  "maxPerInterval": 1000,
}
`;
